Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift

```diff
@@ -93,7 +112,23 @@ extension BrowserViewController: RewardsDataSource {
         }
     }
     
-    func displayString(for url: URL) -> String? {
-        return url.host
+    func pageHTML(for tabId: UInt64, completionHandler: @escaping (String?) -> Void) {
+        guard let tab = tabManager.tabsForCurrentMode.first(where: { $0.rewardsId == tabId }),
+            let webView = tab.webView else {
+                completionHandler(nil)
+                return
+        }
+        
+        let getHtmlToStringJSCall = "document.documentElement.outerHTML.toString()"
+        
+        DispatchQueue.main.async {
+            webView.evaluateJavaScript(getHtmlToStringJSCall, completionHandler: { html, _ in
```

Comment:
we should log `error` here imo, unless this js call can throw a lot on unnecessary errors